### ISSUE TITLE
HDF5. File accessibility. Unable to open file.

### ISSUE THREAD

**Issue by tjhwangxiong** [fmicompbio/TabulaMurisSenisData#19]
library(TabulaMurisSenisData)
> (tms_facs <- TabulaMurisSenisFACS(tissues = "All"))
snapshotDate(): 2024-10-24
see ?TabulaMurisSenisData and browseVignettes('TabulaMurisSenisData') for documentation
loading from cache
Error: rhdf5::H5Fopen(name, flags = flags, fapl = fapl): 
  HDF5. File accessibility. Unable to open file.

Moreover, I have downloaded 'tabula-muris-senis-facs-processed-official-annotations.h5ad' and'tabula-muris-senis-facs-official-raw-obj.h5ad' in the workding directory.

Thanks a lot.

![Image](https://github.com/user-attachments/assets/3653c50f-7c33-4a43-a521-f9027fe99ac4)

**Comment by csoneson** [comment]
Hi @tjhwangxiong - could this be an issue with internet access/proxy settings or similar? I just tried running

```
(tms_facs <- TabulaMurisSenisFACS(tissues = "All"))
```

and it downloads the data and returns a list with a `SingleCellExperiment`, as expected.

**Comment by tjhwangxiong** [comment]
It may be the access issue, while I downloaded successfully.

I used an alternative way as following:
```
rm(list = ls())
library(zellkonverter)
library(SingleCellExperiment)
library(Seurat)
library(scater)
library(patchwork)

#download h5ad files
#https://figshare.com/articles/dataset/Processed_files_to_use_with_scanpy_/8273102?file=23937842
#https://figshare.com/ndownloader/files/23937842
#https://figshare.com/ndownloader/files/23938934

sce <- readH5AD("tabula-muris-senis-facs-processed-official-annotations.h5ad")
# assays
assayNames(sce)
#[1] "X"

# Assign the data from X to counts
assay(sce, "counts") <- assay(sce, "X")

# Remove the X assay to free up memory
assay(sce, "X") <- NULL

assayNames(sce)
#[1] "counts"

logcounts(sce) <- log1p(assay(sce, "counts"))

# Transform to seurat
sce <- as.Seurat(sce)

names(sce@reductions)
#[1] "X_pca"  "X_umap"

sce@reductions[["pca"]] <- sce@reductions[["X_pca"]]
sce@reductions[["X_pca"]] <- NULL
sce@reductions[["umap"]] <- sce@reductions[["X_umap"]]
sce@reductions[["X_umap"]] <- NULL

#Run TSNE
sce <- RunTSNE(sce, dims = 1:ncol(sce@reductions$pca))

saveRDS(sce,file="sce.rds")
```

Anyway, Thanks a lot.

**Comment by csoneson** [comment]
Ok, great that you could solve it. If it's useful, you can also find the code that we used to process the raw data linked in the bottom of our [README](https://github.com/fmicompbio/TabulaMurisSenisData?tab=readme-ov-file).
